### ISSUE TITLE
API not responding within 20 minutes to seemingly simple request

### ISSUE THREAD

**Issue by davidherbert2** [urbanobservatory/uo-api-v1.1#3]
I've sent the following request for all NO2 sensor readings in the Air Quality theme for a 3 hour period.  No response received after 20 minutes.  Have tried it several times this afternoon, and similar requests intermittently over the past few weeks. I am not expecting to receive much data back, however I'm not sure whether this assumption is correct?  Maybe this call is actually returning all the sensor telemetry/diagnostic info, and I should be using something else?  It would be helpful if the API docs gave example returns for the sample requests.  Here's the request:

http://uoweb3.ncl.ac.uk/api/v1.1/sensors/data/json/?starttime=20190523120000&endtime=20190523150000&theme=Air+Quality&data_variable=NO2&bbox_p1_x=-1.7353&bbox_p1_y=54.9484&bbox_p2_x=-1.4298&bbox_p2_y=55.0563

Even adding "last_n_records=100" makes no difference.  Still no response.

My eventual goal is to be able to call the UO API from Ajax requests in a front-end application, possibly linking said requests to the progress of a timeslider, so I need these requests to take a few seconds max.  Should this not be realistic, is it possible to get a more low-level access to the data directly via the database?  The only way I can see otherwise of doing this is doing a bulk daily download at night and ingesting into my own PostGIS instance.  This seems to defeat the purpose of the API, and uses up large amounts of space on a VM anyway.  BTW I see I could get all the relevant sensor names and get the readings for each individually, which seems rather longwinded.

**Comment by geoanorak** [comment]
Thanks David, we'll take a look.  The API has been a bit of an issue for a while with larger calls.

**Comment by nharris172** [comment]
This should now be faster.

**Comment by davidherbert2** [comment]
Yep - can confirm that this request now takes 4-5 seconds to execute.  Thanks Neil.
